### DISCUSSION TITLE
Better understanding for properties.conf

### DISCUSSION THREAD

**Discussion by superit63** [engageub/InternetIncome#129] [Q&A]
Hello,

Here is the config in properties.conf:
Enable the following network configuration if you use proxies
Create a file name proxies.txt and add your proxies in the format protocol://user:pass@ip:port
USE_PROXIES=true

Enable the following network configuration if you use VPN (Openvpn,wireguard etc)
Create a file name vpns.txt and add your vpn configuration as mentioned in the readme file.
USE_VPNS=false

Enable the following network configuration if you have machine with multiple IP address.
Note that this only requires your public IP address
Create a file name multiips.txt and add your public IP address in each line
USE_MULTI_IP=false

Set to true if you would like to use DNS over HTTPS or TCP
USE_DNS_OVER_HTTPS=false

Set to true if your socks5 proxy supports DNS
USE_SOCKS5_DNS=false

Use Custom Networks if you have more than 1000 proxies in proxies.txt file
USE_CUSTOM_NETWORK=false

Set to true if you would like to use tun2proxy instead of tun2socks
USE_TUN2PROXY=false
----------------
Could you give me the example use case that:
1. when do we need to set true on DNS over https or socks5?
2. When we use Tun2proxy instead of tun2socks as default?
3. The direct connection is only changed to true whenever we dont use proxy (I mean we dont use proxy and when the USE_PROXIES= false). 
Thanks

**Comment by engageub** [comment]
Hello,
Please find the answers below.

1. DNS requests are used to get the IP address for a given domain name. They are typically sent over UDP Port 53 and are transmitted in plain text. If you would like to encrypt the DNS request, you will need to use DNS over HTTPS (DoH) or DNS over TLS (DoT). By default the DNS requests go through your VPS instead of your proxy mentioned. If you would like to redirect the DNS requests over your proxy, you can either use DNS over HTTPS/TCP or SOCKS5 proxy.  Use SOCKS5 DNS if your proxy supports UDP protocol.
2. Tun2proxy is an alternative to Tun2socks. Both have similar functionality of redirecting the traffic through proxy. However, tun2socks has problem with default DNS bypass on older kernel versions. Example [play with docker](https://labs.play-with-docker.com/). You can use tun2proxy if tun2socks does not work when using proxies.
3. The test branch supports different network options at the same time. You can run an application on direct connection, proxy, vpn and multi ip at the same time. You can choose which network mode you wish to run the apps on.  By default the script does not start if network mode is not specified in test branch. In main branch, it is different, as the default is direct connection and only one network option works at a time. 

Thank you

**Reply by superit63** [reply]
Hi,
Thanks a lot for your explanation.
1. Related to DNS, I would like to summarize that only enable DNS request if I have HTTPS/Socks5 with enabled UDP. Its optional parameter, I can use or dont need to use it as long as my application need UDP or not. As far as I know, only Mystnode needs UDP protocol
2. Related to Tun2proxy, its simply that I can try with tun2socks as default and if it doesnt work, just change to tun2proxy.
3. If I enable VPN, direct connection and proxy, I will have 1 application (Honeygain as an example) with 3 containers that are running on direct connection, VPN and Proxy environment. Is it right?
Thanks

**Reply by engageub** [reply]
Please find the answers below.

1. HTTP,HTTPS,SOCKS4 do not have UDP protocol, only SOCKS5 supports UDP protocol. If your proxy provider, mentions that UDP is enabled on Socks5 then you can use Socks5 DNS. DNS over HTTPS (DoH) adds an extra layer of security by encrypting DNS queries, preventing them from being sent in plain text. By default, it is not required to make any DNS changes. Only Mystnodes require UDP as of now. However, I got to know that honeygain latest version also makes UDP requests sometimes. The current code, uses older version of honeygain while using proxies.
2. Your understanding is correct about tun2socks and tun2proxy.
3. Yes, that's correct. If you enable 3 different network options, then 3 containers of same app run through different network modes.
